File: sol/118/118.py
class Solution:
    ALL_ROWS = []
    def __init__(self):
        if len(self.ALL_ROWS) == 0:
            self.ALL_ROWS.append([1])
            prev = [1]
            for _ in range(30 - 1):
                prev = [1] + [a + b for a, b in zip(prev[:-1], prev[1:])] + [1]
                self.ALL_ROWS.append(prev)
    def generate(self, numRows: int) -> List[List[int]]:
        return self.ALL_ROWS[:numRows]

File: sol/118/118.cpp
struct Solution {
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        res.push_back(vector<int>(1, 1));
        for (int i = 0; i < numRows - 1; ++i){
            res.push_back(vector<int>(1, 1));
            for (size_t j = 0; j < res[i].size() - 1; ++j){
                res.back().push_back(res[i][j] + res[i][j + 1]);
            }
            res.back().push_back(1);
        }
        return res;
    }
};
